src/cli/serve/serve.js

```diff
@@ -180,10 +182,21 @@ export default function (program) {
         kbnServer.server.log(['info', 'config'], 'Reloaded logging configuration due to SIGHUP.');
       });
 
+      if (shouldStartRepl(opts)) {
+        startRepl(kbnServer);
+      }
+
       return kbnServer;
     });
 }
 
+function shouldStartRepl(opts) {
+  // The kbnWorkerType check is necessary to prevent the repl
+  // from being started multiple times in different processes.
+  // We only want one REPL.
+  return opts.repl && process.env.kbnWorkerType === 'server';
```

Comment:
We should constrain the repl from being used only while in dev mode.

Reply:
Yes. I agree with this 100%. I didn't see a great way to detect that, but I will make it happen.